### PR TITLE
Dev : eoa crate migration into contracts crates

### PR DESCRIPTION
**Pull request by Quentash**
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [X] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #485 

## What is the new behavior?

- Migrated eoa crate into contratcs crate src directory and corrected imports.
-
-

## Does this introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

File: crates/contracts/src/eoa.cairo
Comment (enitrat) [inline crates/contracts/src/eoa.cairo:1]: do we need eoa.cairo -> externally_owned_account.cairo ?
I think we can just add the content of `externally_owned_account` inside `eoa `directly

Comment (Eikix) [reply crates/contracts/src/eoa.cairo:1]: i vote that we have interface.cairo and eoa.cairo inside the eoa folder, to match contract_account

Comment (Eikix) [reply crates/contracts/src/eoa.cairo:1]: wdyt @Quentash @enitrat

Comment (enitrat) [reply crates/contracts/src/eoa.cairo:1]: and re-export them from there like OZ does?

Comment (enitrat) [reply crates/contracts/src/eoa.cairo:1]: > to match contract_account

well, the question would be, do we also need this in CA ?

Comment (Eikix) [reply crates/contracts/src/eoa.cairo:1]: yeah we could do that

> and re-export them from there like OZ does?

Comment (Eikix) [reply crates/contracts/src/eoa.cairo:1]: tbh i don't really mind either way

Comment (Quentash) [reply crates/contracts/src/eoa.cairo:1]: Hmm, I'm not really sure what I gotta do here ^^'
I've moved `externally_owned_account` inside `eoa` directly, but are you suggesting to create a eoa folder inside contracts/src instead ?

**Comment by Eikix** [comment]
Closing in favour of #515 because of immense git conflicts and change of scope of the PR

**Pull request closed without merging**